internal/app/backend/backend_service.go

```diff
@@ -179,43 +176,6 @@ func doFetchMatchesSendResponse(ctx context.Context, proposals []*pb.Match, send
 	return nil
 }
 
-func getHTTPClient(cfg config.View, mmfClients *sync.Map, funcConfig *pb.FunctionConfig_Rest) (*http.Client, string, error) {
```

Comment:
I have second thoughts about the need for this changes. The Synchronizer (unlike backend) only ever needs to connect to one evaluator - so the repeated code between these components is minimal (the mmfclients map is not relevant to synchronizer). For now, the synchronizer can just initialize the connection during its init - and store it on its service struct - unlike the backend that needs one per MMF. 

The current implementation does not actually abstract the cache anyways. In future, I would expect the 'cache' to actually abstract the map from the caller - thus only supplying the caller with connection. The cache would also handle stale connections and reinitialize them if needed etc. For now, we can let this stay as is.